addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/check-launch-dependency/check-launch-dependency.js

```diff
@@ -64,370 +64,386 @@ const successStatuses = ['CREATE_COMPLETE', 'DELETE_COMPLETE', 'UPDATE_COMPLETE'
+      this.state.setKey('DEPENDENCY_TYPE', 'ALB');
+      // Create Stack
+      // eslint-disable-next-line no-return-await
+      return await this.deployStack(requestContext, resolvedVars, stackInput);
     }
+    return null;
+  }
 
-    /**
-    * Method to get role arn for the target aws account
-    *
-    * @param requestContext
-    * @param resolvedVars
-    * @returns {Promise<string>}
-    */
-    async getTargetAccountRoleArn(requestContext, resolvedVars) {
-        const [ALBService] = await this.mustFindServices(['albService']);
-        const { roleArn } = await ALBService.findAwsAccountDetails(requestContext, resolvedVars.projectId);
-        return roleArn;
-    }
-
-    /**
-    * Method to parse the output and key from the stack info
-    *
-    * @param stackInfo
-    * @returns {Promise<{OutputKey: string:, OutputValue: string}[]>}
-    */
-    getStackOutputs(stackInfo) {
-        const details = {};
-        stackInfo.Outputs.forEach(option => {
-            _.set(details, option.OutputKey, option.OutputValue);
-        });
-        return details;
+  /**
+   * Method to deploy ALB stack. The method gets cfn client and calls createStack API
+   *
+   * @param requestContext
+   * @param resolvedVars
+   * @param stackInput
+   * @returns {Promise<>}
+   */
+  async deployStack(requestContext, resolvedVars, stackInput) {
+    const cfn = await this.getCloudFormationService(requestContext, resolvedVars);
+    const response = await cfn.createStack(stackInput).promise();
+    // Update workflow state and poll for stack creation completion
+    this.state.setKey('STACK_ID', response.StackId);
+    return (
+      this.wait(5) // check every 5 seconds
+        // keep doing it for 1*1296000 seconds = 15 days
+        // IMPORTANT: if you change the maxAttempts below or the wait check period of 5 seconds above
+        // then make sure to adjust the error message in "reportTimeout" accordingly
+        .maxAttempts(1296000)
+        .until('shouldResumeWorkflow')
+        .thenCall('onSuccessfulCompletion')
+        .otherwiseCall('reportTimeout')
+      // if anything fails, the "onFail" is called
+    );
+  }
+
+  /**
+   * Method to handle completion of ALB provisioning stack.
+   * Method runs after the stack completion and stores the details in DB
+   *
+   * @param stackOutputs
+   * @returns {Promise<>}
+   */
+  async handleStackCompletion(stackOutputs) {
+    const [requestContext, resolvedVars, stackId, dependencyType] = await Promise.all([
+      this.payloadOrConfig.object(inPayloadKeys.requestContext),
+      this.payloadOrConfig.object(inPayloadKeys.resolvedVars),
+      this.state.string('STACK_ID'),
+      this.state.string('DEPENDENCY_TYPE'),
+    ]);
+    const projectId = resolvedVars.projectId;
+    const [albService] = await this.mustFindServices(['albService']);
+    const awsAccountId = await albService.findAwsAccountId(requestContext, projectId);
+    if (dependencyType === 'ALB') {
```

Comment:
Can `dependencyType` be something else too? What is the rationale behind introducing `DEPENDENCY_TYPE`.

Reply:
This was based on a previous assumption. Have removed this logic and updated the PR